pkg/plugins/gateway/prefixcacheindexer/tree_test.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2024 The Aibrix Team.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prefixcacheindexer
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/vllm-project/aibrix/pkg/utils"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Test_LPRadixCacheE2E(t *testing.T) {
+	cache := NewLPRadixCache(2) // assuming 2 GPUs
+	pods := []*v1.Pod{
+		{ObjectMeta: metav1.ObjectMeta{Name: "p1"}},
+		{ObjectMeta: metav1.ObjectMeta{Name: "p2"}},
+	}
+
+	inputText := "Hello World! What a Good Day! Good Morning! 你好世界！多么美好的一天啊！早上好！"
```

Comment:
This is not directly related to your PR. just want you know some feedback from users. Did you check issue here? https://github.com/vllm-project/aibrix/issues/673 could we make this part pluggable to support string (or chracter base) as well?  We do not necessarily need to support this case in this PR here, just for discussion